examples/Examples/Data/Primitives.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE 
+    MultiParamTypeClasses
+  , TemplateHaskellQuotes 
+#-}
+
+module Examples.Data.Primitives 
+( MyError (..)
+, Username
+, mkUsername
+, unUsername
+, EmailAddress
+, mkEmailAddress
+, unEmailAddress
+, PhoneNumber
+, mkPhoneNumber
+, unPhoneNumber
+) where
+
+------------------------------------------------------------------------------------------------------------------------
+import Data.Validation
+------------------------------------------------------------------------------------------------------------------------
+
+data MyError

```

Comment:
Too keep with vocabulary in RFC this should probably be `ValidationFailures` or similar